nntrainer/tensor/tensor.h

```diff
@@ -504,7 +504,22 @@ class Tensor {
    * @param[in] direction to transpose ex) 0:2:1
    * @retval    Calculated Tensor
    */
-  Tensor transpose(std::string direction) const;
+  Tensor transpose(const std::string &direction) const;
+
+  /**
+   * @brief Transpose Tensor
+   * @param direction to transpose ex) 0:2:1
+   * @param[out] Tensor to saved to
+   * @retval Tensor& reference to the out
+   */
+  Tensor &transpose(const std::string &direction, Tensor &out) const;
+
+  /**
+   * @brief Transpose a tensor and put it back to @a this
+   *
+   * @param direction to transpose ex) 0:2:1
+   */
+  void transpose_i(const std::string &direction);
```

Comment:
i highly recommend you to add a note here, that this is NOT an in-place operation. It just returns the result in-place.

Reply:
infact, unless really needed, lets make this private. This gives a false notion of in-place transpose operation while is it not.

Reply:
Transpose cannot be non-copyable, from behavior point of view, it is what `_i` does. Can you suggest different name?

Reply:
How about allow create `Tensor &Tensor::transpose(const std::string &direction, Tensor &out)`, where out can be same as `this`, and this function can call `transpose_i` internally.

Reply:
Sure, I'll change that way.

Reply:
Done in a new commit. removed `transpose_i`